src/clj/rems/validate.clj

```diff
@@ -29,23 +30,22 @@
         (log/warn "Languages missing from form template" (:form/id template) (pr-str (:form/title template))
                   errors)))))
 
-(defn- valid-organization? [org]
-  (contains? (set (:organizations env)) org))
-
 (defn validate-organizations []
   ;; only warning for now
-  (doseq [form (form/get-form-templates {})]
-    (when-not (valid-organization? (:form/organization form))
-      (log/warn "Unrecognized organization in form: " (pr-str form))))
-  (doseq [resource (db/get-resources {})]
-    (when-not (valid-organization? (:organization resource))
-      (log/warn "Unrecognized organization in resource: " (pr-str resource))))
-  (doseq [license (licenses/get-all-licenses {})]
-    (when-not (valid-organization? (:organization license))
-      (log/warn "Unrecognized organization in license: " (pr-str license))))
-  (doseq [item (catalogue/get-localized-catalogue-items)]
-    (when-not (valid-organization? (:organization item))
-      (log/warn "Unrecognized organization in catalogue item: " (pr-str item)))))
+  (let [organizations (->> (organizations/get-organizations-raw) (map :organization/id) set)
+        valid-organization? (fn [id] (contains? organizations id))]
+    (doseq [form (form/get-form-templates {})]
+      (when-not (valid-organization? (:form/organization form))
+        (log/warn "Unrecognized organization in form:" (pr-str form))))
+    (doseq [resource (db/get-resources {})]
+      (when-not (valid-organization? (:organization resource))
+        (log/warn "Unrecognized organization in resource:" (pr-str resource))))
+    (doseq [license (licenses/get-all-licenses {})]
+      (when-not (valid-organization? (:organization license))
+        (log/warn "Unrecognized organization in license:" (pr-str license))))
+    (doseq [item (catalogue/get-localized-catalogue-items)]
+      (when-not (valid-organization? (:organization item))
```

Comment:
is `(:organization item)` not in the `{:organization/id id}` format? if it is, `valid-organization?` expects something else

Reply:
The database contains strings only. Should db level always translate these or should it rather be the service layer? This is not defined at the moment and we should decide something for #2179 . I'm kind of mixed still. Certainly all the entities in db and service layers have some variation in the conventions, e.g. names of format/mapping/fix-from-db functions too.

Reply:
using the service layer here was a good choice